src/Graph.ts

```diff
@@ -372,6 +394,15 @@ export default class Graph {
 				entryModules.forEach(entryModule => {
 					entryModule.chunk.generateEntryExports(entryModule);
 				});
+				if (preserveModules) {
+					// preserve the links between modules
+					// (normally lost when combining modules into chunks)
+					orderedModules.forEach(module => {
+						if (entryModules.indexOf(module) === -1) {
+							module.chunk.generateEntryExports(module, preserveModules);
+						}
+					});
+				}
```

Comment:
This changes things a little unfortunately. I was under impression that this approach was based on "treating all modules as entry points" which would mean exposing all their exports. But it sounds like we still want tree shaking of exports to apply here.

As such, we're not really treating all modules as entry points, so perhaps my previous code suggestions were a little misleading as that was my impression.

What happens if this entire section is removed along with the `isEntryPoint` being set in the analyseExecution method? That seems like it would get us closer to the consistent conceptual model with this here.

Reply:
I tried that but the code that checks `if (chunk.entryModule) {` or `if (chunk.isEntryModuleFacade) {` is now misbehaving. I tried fixing those, but then the object diffs were not making any sense to me:
![capture](https://user-images.githubusercontent.com/602423/36461260-53abda8c-1671-11e8-883e-f87f3c0a6dc1.PNG)

Reply:
You might get that if there were no entry points at all... Perhaps in the case of no entry points all modules should be treated as entry points, but if there are entry points, then nothing else should be treated as an entry point - is that the sort of workflow you're going for here?

Reply:
No. There is always at least one entry point for this feature.